eth/stagedsync/stage_execute.go

```diff
@@ -461,13 +461,13 @@ Loop:
 			if !errors.Is(err, context.Canceled) {
 				logger.Warn(fmt.Sprintf("[%s] Execution failed", logPrefix), "block", blockNum, "hash", block.Hash().String(), "err", err)
 				if cfg.hd != nil {
-					cfg.hd.ReportBadHeaderPoS(blockHash, block.ParentHash())
+					cfg.hd.ReportBadHeaderPoS(blockHash, block.ParentHash() /* lastValidAncestor */)
 				}
 				if cfg.badBlockHalt {
 					return err
 				}
+				u.UnwindTo(blockNum-1, blockHash /* badBlock */)
 			}
-			u.UnwindTo(blockNum-1, block.Hash())
```

Comment:
Do we still want to unwind the block ?  Because it will need to be re-processed.

i.e. do this

```go
u.UnwindTo(blockNum-1, libcommon.Hash{})
```

Reply:
ok